examples/sendoauth2.phps

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/**
+ * This example uses the SendOauth2 wrapper to support OAuth2 (and Basic) authentication for both Microsoft
+ * 365 Exchange email and Google Gmail.
+ * Client secrets and X.509 certificates are supported for Exchange. Client secrets are supported for Gmail.
+ * Authorization_code grant flow and client_credentials (i.e. application) grant flow for SMTP are supported for
+ * Exchange. Authorization_code grant flow is supported for Gmail.
+ * Appropriate scopes (client permissions) and 'provider' overrides are added automatically.
+ *
+ * Install with Composer from the decomplexity/SendOauth2 repo.
+ *
+ * SendOauth2 can be also be invoked using less (or even no) arguments - see the repo for details.
+ *
+ * Needs PHPMailer >=6.6.0 that added support for oauthTokenProvider
+ *
+ * (The next release [V4] of the wrapper will replace TheLeague's Google provider by Google's own GoogleOauthClient;
+ * this will provide support for Google's version of client credentials (Service Accounts) and client certificates)
+ */
+
+//Set the wrapper namespace
+namespace decomplexity\SendOauth2;
+
+//Import PHPMailer classes into the global namespace
+//These must be at the top of your script, not inside a function
+use PHPMailer\PHPMailer\PHPMailer;
+use PHPMailer\PHPMailer\SMTP;
+use PHPMailer\PHPMailer\Exception;
+
+//Load Composer's autoloader
+require 'vendor/autoload.php';
+
+//Create an instance; passing `true` enables exceptions
+$mail = new PHPMailer(true);
+
+try {
+    //Server settings
+    $mail->SMTPDebug = SMTP::DEBUG_SERVER;                      //Enable verbose debug output
+    $mail->isSMTP();                                            //Send using SMTP
+    $mail->Host       = 'smtp.office365.com';                   //Set the SMTP server (smtp.gmail.com for Gmail)
+    $mail->SMTPAuth   = true;                                   //Enable SMTP authentication
+    $mail->Username   = 'user@example.com';                     //SMTP username
+    $mail->SMTPSecure = PHPMailer::ENCRYPTION_STARTTLS;         //Enable implicit TLS encryption
+    $mail->Port       = 587;                                    //TCP port to connect to
+    $mail->AuthType   = 'XOAUTH2';                              // Set AuthType to use XOAUTH2
+
+    //Sender and recipients
+    $mail->setFrom('from@example.com', 'Mailer');               // 'Header' From address with optional sender name
+    $mail->addAddress('joe@example.net', 'Joe User');           //Add a recipient
+    $mail->addAddress('ellen@example.com');                     //Name is optional
```

Comment:
Since the purpose of this example is purely to demonstrate how to use your auth wrapper, I suggest removing anything that is covered in other examples, such as extra recipients, attachments, etc.